posix/signal.c

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+
+#include <openenclave/corelibc/errno.h>
+#include <openenclave/corelibc/signal.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <openenclave/corelibc/unistd.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/posix/lock.h>
+#include <openenclave/internal/posix/raise.h>
+#include <openenclave/internal/print.h>
+#include <openenclave/internal/thread.h>
+#include <openenclave/internal/trace.h>
+#include "posix_t.h"
+
+// Poll uses much of the infrastructure from epoll.
+
+static void _handle_ignore(int signum);
+
+static struct oe_sigaction _actions[__OE_NSIG] = {{{0}}};
+
+static oe_sighandler_t _default_actions[__OE_NSIG] = {
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore,
+    _handle_ignore, _handle_ignore, _handle_ignore, _handle_ignore};
+
+static oe_spinlock_t _lock = OE_SPINLOCK_INITIALIZER;
+
+static void _handle_ignore(int signum)
+{
+    (void)signum;
+}
+
+static void _handle_error(int signum)
+{
+    (void)signum;
```

Comment:
Shouldn't we be propagating the errno set on the host in this case at least? It's also worth tracing the result.

Reply:
Removed singal feature from this PR.